lib/ProductOpener/Products.pm

```diff
@@ -2828,14 +2829,25 @@ my %actions_urls = (
 	add_packager_codes_image => "#packager_codes",
 	add_labels => "#labels",
 	add_countries => "#countries",
+	# this is for web rendering so source is web
+	report_product_to_nutripatrol => "$nutripatrol_url/flag/product/?barcode=PRODUCT_CODE&source=web&flavor=$flavor"
 );
 
 sub product_action_url ($code, $action) {
 
-	my $url = "/cgi/product.pl?type=edit&code=" . $code;
-
+	my $url;
 	if (defined $actions_urls{$action}) {
-		$url .= $actions_urls{$action};
+		my $action_url = $actions_urls{$action};
+		if ($action_url =~ /^#/) {
```

Comment:
The "edit_product" action has an empty url "" because before we always started with the edit form URL. So we need a change like this:

```suggestion
		if (($action eq '') or (($action_url =~ /^#/)) {
```

Reply:
well spotted !